src/c++/vdo/base/vdo.c

```diff
@@ -500,7 +940,48 @@ static int initialize_vdo(struct vdo *vdo, struct device_config *config,
 	vdo_initialize_completion(&vdo->admin.completion, vdo, VDO_ADMIN_COMPLETION);
 	init_completion(&vdo->admin.callback_sync);
 	mutex_init(&vdo->stats_mutex);
-	result = read_geometry_block(vdo);
+
+	// Check to see whether we should format the device
+	result = vdo_is_formatted(vdo, &formatted);
+	if (result != VDO_SUCCESS) {
+		*reason = "Could not determine whether layout is empty";
+		return result;
```

Comment:
This is a problem. If we get an error here, we won't format, but we won't start, either. We shoul be cler about what the error is and what the user can do about it.

It looks like the only real failures are a) an allocation failure for the read buffer; or b) an I/O failure reading the geometry block. It's better to read the geometry block first, reporting errors like we always did, and then check if we're formatted, once we have the data and the format check can't fail.